src/coreclr/jit/morph.cpp

```diff
@@ -14223,6 +14223,47 @@ GenTree* Compiler::fgMorphSmpOpOptional(GenTreeOp* tree)
             }
             break;
 
+#if defined(FEATURE_HW_INTRINSICS) && defined(TARGET_XARCH)
+        case GT_HWINTRINSIC:
+        {
+            GenTreeHWIntrinsic* hw = tree->AsHWIntrinsic();
+            switch (hw->gtHWIntrinsicId)
+            {
+                case NI_SSE_Xor:
+                case NI_SSE2_Xor:
+                case NI_AVX_Xor:
+                case NI_AVX2_Xor:
+                {
+                    // Transform XOR(X, 0) to X for vectors
+                    GenTree* op1 = hw->gtGetOp1();
+                    GenTree* op2 = hw->gtGetOp2();
+                    if (!gtIsActiveCSE_Candidate(tree))
+                    {
+                        if (op1->IsIntegralConstVector(0) && !gtIsActiveCSE_Candidate(op1))
```

Comment:
I think that `IsIntegralConstVector` will only work for integer vectors, so all floating point vectors will be rejected here.

Reply:
Xor is not used in `Equals` for floats and doubles so it's not a big deal

but it currently handles this (Xor for floats) just fine:
```csharp
static Vector128<float> Foo(Vector128<float> v) => 
    Sse.Xor(v, Vector128.Create(0).AsSingle());
```

Reply:
for zero float it might be tricky as e.g. `-0.0` can't be used in this optimization so is not worth the effort

Reply:
Why would it be tricky? You can account for `-0.0` by checking the bitwise (rather than floating-point) value is `0`.

Speaking of `0.0` vs `-0.0`, it looks like there might be existing bugs in `IsFPZero` and `IsSIMDZero` since they don't take this into account.